### PR TITLE
1D vectors instead of nD in Fully-Connected layer

### LINKED ISSUE #38: Use 1D vector instead of nD for matrices

### PR DESCRIPTION
**Pull request by NeiroYT**
Closes #38

### CODE REVIEW

File: include/perf/benchmarking.hpp [hunk 1]
```diff
@@ -56,6 +55,32 @@ double elapsed_time_omp_avg(const size_t iters, Function&& func,
   return (end - start) / iters;
 }
 
+template <typename ThroughputContainerType, typename DurationType,
```
Comment (allnes) [inline include/perf/benchmarking.hpp:58]: please remove content from different PRs

Comment (NeiroYT) [reply include/perf/benchmarking.hpp:58]: My mistake

Comment (NeiroYT) [reply include/perf/benchmarking.hpp:58]: How can i branch from main now

Comment (aobolensk) [reply include/perf/benchmarking.hpp:58]:
```
git checkout main
git pull origin main
git checkout -b <branch-name>
# add things for this feature
git commit ...
git push origin HEAD
```

**Pull request closed without merging**